### PR TITLE
[BGP] add `test_bgp_gr_helper_all_routes_preserved`

### LINKED ISSUE #3974: [New testcase] Verify bgp routes after graceful restart
Existing BGP GR testcase test_bgp_gr_helper_routes_perserved only checks for default route presence during graceful-restart.
This has two limitations:

not support topologies that have no default route(t0-backend and t1-backend)
doesn't check other generic routes received/learned from the neighbor experiencing a BGP restart.

### PR DESCRIPTION
**Pull request by lolyu**
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

Please provide following information to help code review process a bit easier:
-->
### Description of PR
<!--
- Please include a summary of the change and which issue is fixed.
- Please also include relevant motivation and context. Where should reviewer start? background context?
- List any dependencies that are required for this change.
-->

Summary:
Fixes #3974

### Type of change

<!--
- Fill x for your type of change.
- e.g.
- [x] Bug fix
-->

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)


### Back port request
- [ ] 201911

### Approach
#### What is the motivation for this PR?
Existing BGP GR testcase `test_bgp_gr_helper_routes_perserved` only checks for default route presence during graceful-restart.
This has two limitations:
1. not support topologies that have no default route(`t0-backend` and `t1-backend`)
2. doesn't check other generic routes received/learned from the neighbor experiencing a BGP restart.

Signed-off-by: Longxiang Lyu <lolv@microsoft.com>


#### How did you do it?
* Add new test case `bgp/test_bgp_gr_helper.py::test_bgp_gr_helper_all_routes_preserved` to test that those routes received/learned from one neighbor will be kept during the BGP restart, and they should all be in the `STALE` state. 

#### How did you verify/test it?
```
bgp/test_bgp_gr_helper.py::test_bgp_gr_helper_routes_perserved PASSED                                                                                                                                                                                                                       [ 50%]
bgp/test_bgp_gr_helper.py::test_bgp_gr_helper_all_routes_preserved PASSED                                                                                                                                                                                                                   [100%]

=================================================================================================================================== 2 passed in 463.38 seconds ====================================================================================================================================
```
#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation 
<!--
(If it's a new feature, new test case)
Did you update documentation/Wiki relevant to your implementation?
Link to the wiki page?
-->

### CODE REVIEW

File: tests/bgp/test_bgp_gr_helper.py [hunk 1]
```diff
@@ -115,3 +120,112 @@ def test_bgp_gr_helper_routes_perserved(duthosts, rand_one_dut_hostname, nbrhost
...
+            cmd = duthost.get_vtysh_cmd_for_namespace(bgp_cmd_ipv4, namespace)
+            routes.update(json.loads(duthost.shell(cmd, verbose=False)['stdout'])["routes"])
+            cmd = duthost.get_vtysh_cmd_for_namespace(bgp_cmd_ipv6, namespace)
+            routes.update(json.loads(duthost.shell(cmd, verbose=False)['stdout'])["routes"])
+        return routes
+
+    def _get_learned_bgp_routes_from_neighbor(duthost, bgp_neighbor):
+        """Get all learned routes from the BGP neighbor."""
+        if is_ipv4_address(unicode(bgp_neighbor)):
+            cmd = "vtysh -c 'show bgp ipv4 neighbor %s routes json'" % bgp_neighbor
```
Comment (neethajohn) [inline tests/bgp/test_bgp_gr_helper.py:151]: Will this cmd work on multi-asic?

File: tests/bgp/test_bgp_gr_helper.py [hunk 2]
```diff
@@ -115,3 +120,112 @@ def test_bgp_gr_helper_routes_perserved(duthosts, rand_one_dut_hostname, nbrhost
...
+        else:
+            cmd = "vtysh -c 'show bgp ipv6 neighbor %s routes json'" % bgp_neighbor
+        cmd_result = json.loads(duthost.shell(cmd, verbose=False)["stdout"])
+        return cmd_result["routes"]
+
+    def _verify_prefix_counters_from_neighbor_after_graceful_restart(duthost, bgp_neighbor):
+        """Verify that all routes received from neighbor are stale after graceful restart."""
+        if is_ipv4_address(unicode(bgp_neighbor)):
+            cmd = "vtysh -c 'show bgp ipv4 neighbor %s prefix-counts json'" % bgp_neighbor
```
Comment (neethajohn) [inline tests/bgp/test_bgp_gr_helper.py:160]: Will this cmd work on multi-asic?

File: tests/bgp/test_bgp_gr_helper.py (hunk 1 above)
Comment (lolyu) [reply tests/bgp/test_bgp_gr_helper.py:152]: updated, will check on all namespaces.

**Comment by lolyu** [comment]
updated, @neethajohn
```
bgp/test_bgp_gr_helper.py::test_bgp_gr_helper_routes_perserved PASSED                                                                                                                                                                                                                       [ 50%]
bgp/test_bgp_gr_helper.py::test_bgp_gr_helper_all_routes_preserved PASSED                                                                                                                                                                                                                   [100%]

=================================================================================================================================== 2 passed in 311.80 seconds ====================================================================================================================================
```

**Review by neethajohn — APPROVED** [review]

**Review by abdosi — APPROVED** [review]

**Pull request merged**